src/hotspot/share/classfile/verificationType.hpp

```diff
@@ -162,7 +162,7 @@ class VerificationType {
       // to discriminate between oops and primitives.
       return VerificationType((uintptr_t)sh);
   }
-  static VerificationType uninitialized_type(u2 bci)
+  static VerificationType uninitialized_type(int bci)
```

Comment:
I'd rather have the caller make sure this is a u2.  If we use int here then we need to make sure the shift below doesn't overflow.

Reply:
VerificationType's constructor is uintptr_t so int << 8 shouldn't overflow, should it? Although it might change sign, which we don't warn about.  I'll change it and make its caller (bci's are passed and used as int everywhere even though they're u2), does a checked_cast<u2>.

Reply:
uintptr_t would be 32-bits on a 32-bit platform, right?

Reply:
I keep forgetting there's 32 bits, ok.